### PR TITLE
asserts: include the assertion timestamp in error message when outside of signing key validity range

### PR DESCRIPTION
**Pull request by bboozzoo**
We occasionally get the following backtrace in the tests:

```
----------------------------------------------------------------------
PANIC: api_users_test.go:762: userSuite.TestPostCreateUserFromAssertionNoModel

... Panic: cannot add test assertions: model assertion timestamp outside of signing key validity (key valid since "2021-02-23 11:10:41 +0000 UTC") (PC=0x45CF4A)

/snap/go/2117/src/runtime/panic.go:491
  in gopanic
/home/runner/work/snapd/snapd/src/github.com/snapcore/snapd/overlord/assertstate/assertstatetest/add_many.go:38
  in AddMany
api_users_test.go:776
  in userSuite.TestPostCreateUserFromAssertionNoModel
/snap/go/2117/src/reflect/value.go:302
  in Value.Call
/snap/go/2117/src/runtime/asm_amd64.s:2337
  in goexit
OOPS: 439 passed, 1 skipped, 1 PANICKED
--- FAIL: Test (9.62s)
FAIL
coverage: 86.4% of statements
FAIL	github.com/snapcore/snapd/daemon	9.786s
```

Adding a timestamp of the assertion, will help it to debug the problem further.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Looks sensible to me but let's see what what Samuele  thinks

**Review by stolowski — APPROVED** [review]
Looks fine to me.

**Review by anonymouse64 — APPROVED** [review]

File: asserts/database.go [hunk 1]
```diff
@@ -725,7 +725,8 @@ func CheckTimestampVsSigningKeyValidity(assert Assertion, signingKey *AccountKey
 			if !signingKey.Until().IsZero() {
 				until = fmt.Sprintf(" until %q", signingKey.Until())
 			}
-			return fmt.Errorf("%s assertion timestamp outside of signing key validity (key valid since %q%s)", assert.Type().Name, signingKey.Since(), until)
+			return fmt.Errorf("%s assertion timestamp %q outside of signing key validity (key valid since %q%s)",
```
Comment (pedronis) [inline asserts/database.go:728]: why %q instead of %v ?

**Review by pedronis — COMMENTED** [review]
looks reasonable but not sure about the formatting

File: asserts/database.go (hunk 1 above)
Comment (bboozzoo) [reply asserts/database.go:728]: Used the same formatting as for the since/until timestamps.

Comment (pedronis) [reply asserts/database.go:728]: ah,  I missed that, sorry,  consistency is good

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**